hybrasyl/Objects/User.cs

```diff
@@ -2305,586 +2145,103 @@ public override void RegenerateMp(double mp, Creature regenerator = null)
-                                    SourceId = this.Id,
-                                    Speed = (short)castObject.Effects.Animations.OnCast.Target.Speed,
-                                    TargetId = target.Id,
-                                    TargetAnimation = castObject.Effects.Animations.OnCast.Target.Id
-                                };
-                                Enqueue(effectAnimation.Packet());
-                                SendAnimation(effectAnimation.Packet());
-                            }
-
-                        }
-                        else
-                        {
-                            //var formula = damage.Formula;
-                        }
-                        CastingCost(castObject);
+                        Logger.Info($"UseCastable: {Name} casting {castObject.Name} - removing status {status}");
+                        RemoveStatus(applyStatus.Icon);
                     }
-                }
-
-                //TODO: DRY
-                Motion motion;
-
-                try
-                {
-                    motion = castObject.Effects.Animations.OnCast.Player.SingleOrDefault(x => x.Class.Contains((Class)Class));
-                }
-                catch (InvalidOperationException)
-                {
-                    motion = castObject.Effects.Animations.OnCast.Player.FirstOrDefault(x => x.Class.Contains((Class)Class));
-
-                    Logger.ErrorFormat("{1}: contains more than one motion for a class definition, using first one found!", castObject.Name);
-                }
-
-                var sound = new ServerPacketStructures.PlaySound { Sound = (byte)castObject.Effects.Sound.Id };
+                    else
+                        Logger.Warn($"UseCastable: {Name} casting {castObject.Name} - failed to remove status {status}, does not exist!");
 
-                if (motion != null)
-                {
-                    var playerAnimation = new ServerPacketStructures.PlayerAnimation()
-                    {
-                        Animation = (byte)motion.Id,
-                        Speed = (ushort)(motion.Speed / 5), //handles the speed offset in this specific packet.
-                        UserId = Id
-                    };
-                    Enqueue(playerAnimation.Packet());
-                    SendAnimation(playerAnimation.Packet());
                 }
-                Enqueue(sound.Packet());
-                PlaySound(sound.Packet());
-                //this is an attack skill
-            }
-            else
-            {
-                //need to handle scripting
-            }
-        }
 
-        public List<Creature> PossibleTargets(Castable castable)
-        {
-            List<Creature> actualTargets = new List<Creature>();
+                if (castObject.Effects.Animations.OnCast.Target == null) continue;
+                Effect(castObject.Effects.Animations.OnCast.Target.Id, castObject.Effects.Animations.OnCast.Target.Speed);

```

Comment:
Effect() only casts on the user that cast the spell, not on the target.

Reply:
Fixed, nice catch.